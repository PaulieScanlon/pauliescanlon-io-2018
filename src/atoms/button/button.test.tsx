import * as React from "react";
import { shallow } from "enzyme";

import { Button } from "./";

describe("<Button />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Button>Hooray</Button>);
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
