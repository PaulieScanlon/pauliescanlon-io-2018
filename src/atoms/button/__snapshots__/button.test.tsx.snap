// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button>
    Hooray
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hooray",
      "className": "css-57ccwp-Button-button e8b2jnb0",
    },
    "ref": null,
    "rendered": "Hooray",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hooray",
        "className": "css-57ccwp-Button-button e8b2jnb0",
      },
      "ref": null,
      "rendered": "Hooray",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
